themes/bootstrap3/templates/search/versions.phtml

```diff
@@ -0,0 +1,108 @@
+
+  $this->render('search/results-scripts.phtml');
+
+  // Load only if list view parameter is NOT full:
+  if ($this->params->getOptions()->getListViewOption() !== 'full') {
+      $this->headScript()->appendFile("record.js");
+      $this->headScript()->appendFile("embedded_record.js");
+  }
+?>
+
+<div class="<?=$this->layoutClass('mainbody')?>">
+  <?php $recordTotal = $this->results->getResultTotal(); ?>
+  <?=$this->flashmessages()?>
+  <div class="search-header hidden-print">
+    <div class="search-stats">
+      <?php if ($recordTotal > 0): ?>
+        <?=$this->context()->renderInContext('search/controls/showing.phtml', ['recordTotal' => $recordTotal]) ?>
+        <a class="search-filter-toggle visible-xs" href="#search-sidebar" data-toggle="offcanvas" title="<?=$this->transEsc('sidebar_expand') ?>">
+          <?=$this->transEsc('Refine Results') ?>
+        </a>
+      <?php else: ?>
+        <h2><?=$this->transEsc('nohit_heading')?></h2>
+      <?php endif; ?>
+    </div>
+
+    <?php if ($recordTotal > 0): ?>
+      <div class="search-controls">
+        <?=$this->render('search/controls/limit.phtml')?>
+        <?=$this->render('search/controls/sort.phtml')?>
+        <?=$this->render('search/controls/view.phtml')?>
+      </div>
+    <?php endif; ?>
+  </div>
+  <?php /* End Listing Options */ ?>
+
+  <?php if ($recordTotal < 1): ?>
+    <p>
+      <?php
+        $emptyMessage = $this->slot('empty-message')->get(
+          $this->translate('nohit_lookfor_html', ['%%lookfor%%' => $this->transEsc('Versions')])
+        );
+        $this->layout()->srmessage = $emptyMessage;
+        echo $emptyMessage;
+      ?>
+    </p>
+    <?php if (isset($this->parseError)): ?>
+      <p class="alert alert-danger"><?=$this->transEsc('nohit_parse_error')?></p>
+    <?php endif; ?>
+  <?php else: ?>
+    <form id="search-cart-form" method="post" name="bulkActionForm" action="<?=$this->url('cart-searchresultsbulk')?>" data-lightbox data-lightbox-onsubmit="bulkFormHandler">
+      <?=$this->context($this)->renderInContext('search/bulk-action-buttons.phtml', ['idPrefix' => ''])?>
+    </form>
+    <?=$this->render('search/list-' . $this->params->getView() . '.phtml')?>
+    <?=$this->context($this)->renderInContext('search/bulk-action-buttons.phtml', ['idPrefix' => 'bottom_', 'formAttr' => 'search-cart-form'])?>
+    <?=$this->paginationControl($this->results->getPaginator(), 'Sliding', 'search/pagination.phtml', ['results' => $this->results, 'options' => isset($this->paginationOptions) ? $this->paginationOptions : []])?>
+    <?=$this->context($this)->renderInContext('search/controls/results-tools.phtml', ['results' => $this->results])?>
+  <?php endif; ?>
+</div>
+<?php /* End Main Listing */ ?>
+
+<?php /* Narrow Search Options */ ?>
+<div class="<?=$this->layoutClass('sidebar')?>" id="search-sidebar">
+  <?php foreach ($this->results->getRecommendations('side') as $index => $current): ?>
```

Comment:
Is there a particular reason why you've kept side recommendations here, but have removed top recommendations above?

Reply:
The train of thought went like this: "Maybe it's better to not display recommendations for versions. Oh, we need side recommendations for SideFacets". I suppose in the end it doesn't make sense to limit like this since there could be TopFacets etc. too.

Reply:
Ultimately I think we may want a mechanism to configure different recommendations for Versions than for regular search results, but I'm fine with starting with the default settings and worrying about the details later. (It is possible to override recommendation settings through request parameters, so we could easily support this by manipulating the request in the versionsAction -- there's an existing mechanism that was built for use by the Combined search results that could be leveraged for this). Just a matter of figuring out the best place to have the configuration, if we need it at all.

Reply:
I started working on it and realized that we already have a mechanism. Since the versions search uses the WorkKeys handler, you can just override the defaults in [TopRecommendations] and [SideRecommendations]. I had to tweak RecommendListener a bit to be able to set recommendations for a handler to false (did I miss an obvious existing method?).